File: C++/Mlecz.cpp
#include <iostream>
#include <windows.h>
#include "Mlecz.h"

Mlecz::Mlecz(int x, int y, Swiat* swiat) : Roslina(x, y, 0, 0, IDMLECZ, swiat)
{
}
Mlecz::Mlecz(int x, int y, bool dz, Swiat* swiat) : Roslina(x, y, dz, 0, 0, IDMLECZ, swiat)
{
}

void Mlecz::rysowanie()
{
	HANDLE col;
	col = GetStdHandle(STD_OUTPUT_HANDLE);
	SetConsoleTextAttribute(col, KOLORMLECZ);
	std::cout << "M";
}

std::string Mlecz::nazwa()
{
	return "M";
}

void Mlecz::stworz_nowe_Roslina(int x, int y, Swiat* sw)
{
	Organizm* org = new Mlecz(x, y, true, sw);
}

int Mlecz::Ileprob()
{
	return 3;
}

Mlecz::~Mlecz()
{
}

File: Python/Mlecz.py
from Roslina import Roslina
from Organizm import Organizm

class Mlecz(Roslina):
    def __init__(self, x, y, dz, swiat):
        super().__init__(x, y, dz, 0, 0, swiat.IDMLECZ, swiat)

    def rysowanie(self):
        return "#eeea18"

    def nazwa(self):
        return "Mlecz"

    def stworz_nowe_roslina(self, x, y, sw):
        org = Mlecz(x, y, True, sw)

    def ile_prob(self):
        return 3
